executor/src/witgen/query_processor.rs

```diff
@@ -135,7 +135,7 @@ where
                     }
                 } else {
                     Err(IncompleteCause::ExpressionEvaluationUnimplemented(
-                        "Cannot evaluate arrays or next references.".to_string(),
+                        "Cannot evaluate arrays.".to_string(),
```

Comment:
Are these changes needed here?

Reply:
Yes, otherwise the machine doesn't work, but they are also in #735, so if we merge that first they'll go away.